src/DotNetLightning.Core/Channel/CommitmentToLocalExtension.fs

```diff
@@ -0,0 +1,131 @@
+namespace DotNetLightning.Channel
+
+open System
+open NBitcoin
+open NBitcoin.BuilderExtensions
+open DotNetLightning.Utils
+open DotNetLightning.Utils.SeqConsumer
+open DotNetLightning.Crypto
+
+open ResultUtils
+open ResultUtils.Portability
+
+type CommitmentToLocalParameters = {
+    ToSelfDelay: BlockHeightOffset16
+    LocalDelayedPubKey: DelayedPaymentPubKey
+}
+    with
+    static member TryExtractParameters (scriptPubKey: Script): Option<CommitmentToLocalParameters> =
```

Comment:
This function needs a unit test. Or property test parameterized by two items to extract.

Reply:
I've added a unit test for this.